src/ra_utility.h

```diff
@@ -157,6 +157,32 @@ template<typename Enum, class = std::enable_if_t<std::is_enum_v<Enum>>> _CONSTAN
 operator&=(_Inout_ Enum& a, _In_ Enum b) noexcept { return (a = a & b); }
 
 } // namespace bitwise_ops
+
+// So we don't have to #define _CRT_SECURE_NO_WARNINGS, C99/11 style (no-alias optimization)
+// To use instead of std::fopen for unique_ptr
+template<typename CharT, typename = std::enable_if_t<is_char_v<CharT>>>
+_NODISCARD FILE* __cdecl fopen_s(_In_z_ const CharT* __restrict filename,
+                                 _In_z_ const CharT* __restrict mode,
+                                 _In_opt_     FILE*  __restrict streamptr = nullptr)
```

Comment:
Somewhat like C11's fopen_s. This is how it looks like in C11...
```
errno_t fopen_s(FILE *restrict *restrict streamptr, 
                const char *restrict filename, 
                const char *restrict mode);
```

I changed it a bit to suit our purposes better. The MSVC versions are still called.